### PR TITLE
Opendistro 0.10 Use inner channel when current channel is not netty or direct

### PR DESCRIPTION
**Pull request by sujithvm**
Fix: If transport channel is not netty or direct, check if getInnerChannel is implemented using reflection and use that channel.

### CODE REVIEW

**Comment by hardik-k-shah** [comment]
Code change looks good. will wait for combine test result post CI changes.

File: .github/workflows/ci.yml [hunk 1]
```diff
@@ -29,5 +29,21 @@ jobs:
       - name: Test
```
Comment (hardik-k-shah) [inline .github/workflows/ci.yml:29]: name: Test security-ssl

File: .github/workflows/ci.yml [hunk 2]
```diff
@@ -29,5 +29,21 @@ jobs:
...
         run: mvn test
 
-      - name: Install
+      - name: Install security-ssl
         run: mvn clean install -DskipTests
```
Comment (hardik-k-shah) [inline .github/workflows/ci.yml:33]: can we merge with above section?

**Review by debjanibnrj — APPROVED** [review]
LGTM!

**Pull request merged**